app/helpers/local_time_helper.rb

```diff
@@ -67,4 +53,33 @@ def extract_options_and_value(options, value_key = nil)
         [ {}, options ]
       end
     end
+
+    def find_time_format(format)
+      if format.is_a?(Symbol)
+        find_time_format_by_name(format)
+      else
+        [format.presence || LocalTime.default_time_format, nil]
+      end
+    end
+
+    def find_time_format_by_name(format)
+      if format12 = i18n_time_or_date_format(format)
+        format24 = i18n_time_or_date_format("#{format}_24h")
+      elsif fmt = ruby_time_or_date_format(format)
+        format12 = fmt.is_a?(Proc) ? nil : fmt
+        format24 = ruby_time_or_date_format(:"#{format}_24h")
+      else
+        format12, format24 = nil, nil
+      end
+
+      [format12.presence || LocalTime.default_time_format, format24.presence]
```

Comment:
Would it be possible to move the defaults to `i18n_time_or_date_format` and `ruby_time_or_date_format` and to make this method a bit easier to read? If so we could remove the `||` here and the `else` above.

Reply:
So `i18n_time_or_date_format` and `ruby_time_or_date_format` are actually serving a dual purpose. They return the format, but they're also specifying whether we should use an `i18n` or `ruby` strategy for looking up the `_24h` counterpart.

Moving the defaults there would interfere with that logic. Which simply indicates this _does_ need a rewrite.

Clean this up in https://github.com/basecamp/local_time/pull/135/commits/4aba3fb1e0f08d6293623cb717161595d10e32b5, let me know what you think.

Reply:
This reads much better now, thank you!